helper/terraformtype/helper/schema/type_schema.go

```diff
@@ -139,6 +153,21 @@ func NewSchemaInfo(cl *ast.CompositeLit, info *types.Info) *SchemaInfo {
 		result.Schema.ValidateFunc = func(interface{}, string) ([]string, []error) { return nil, nil }
 	}
 
+	if kvExpr := result.Fields[SchemaFieldElem]; kvExpr != nil && astutils.ExprValue(kvExpr.Value) != nil {
```

Comment:
Same here -- I'm curious about the potential implementation.

Reply:
Sorry I didn't get your point. This part is actually filling in the `SchemaInfo.Schema` with whatever defined in code under analysis. Then user of the `SchemaInfo` could have a nearly complete (depends on our implementation) `tfschema.Schema` as it is in provider code. This can somehow ease the implementation of other rules. E.g. for `S001` instead of checking at ast level, we can just check it via dereferencing the `Elem` in the returned `SchemaInfo.Schema` to see whether it is `nil` or not. (BTW: I find current `S001` doesn't check nilness of that)